src/compiler/types.ts

```diff
@@ -612,7 +612,12 @@ namespace ts {
     export const enum RelationComparisonResult {
         Succeeded = 1, // Should be truthy
         Failed = 2,
-        FailedAndReported = 3

```

Comment:
I'd suggest turning the three values into three distinct bit fields. Then you don't have to first & with a mask, you can just directly test the bit(s) you're interested in.

Reply:
I mean, I I could also say that "Succeeded = 1 << 0", "Failed = 1 << 1", and "Reported = 1 << 2". (So "failed and reported" sets two bits)

Reply:
Sure, whatever works best, as long as it gets rid of the need to first mask and then compare.

Reply:
@ahejlsberg done.